thumbnail/thumbnail_magick.go

```diff
@@ -224,6 +249,8 @@ func detectImageFormat(bytes []byte) int {
 		return FORMAT_WEBP
 	} else if isBMP(bytes) {
 		return FORMAT_BMP
+	} else if isHEIC(bytes) {
```

Comment:
do we need to change `isFormatTransparent` method as well?

Reply:
HEIF specification support transparency, but ImageMagick implementation does not support HEIC transparency.
Therefore, if we include HEIF  in isFormatTransparent, the background color won't work in image compositing.

Reply:
Hi yoya san, could you check https://github.com/strukturag/libheif/issues/56. It seems supporting alpha channel with proper LibHeif version. I could be wrong, correct me. Thanks.

Reply:
ImageMagick doesn't pass an alpha channel to libheif, so it can't handle transparency at the moment.
- ImageMagick passes only 3 channels Y, Cb, Cr
  - https://github.com/ImageMagick/ImageMagick/blob/7.0.10-17/coders/heic.c#L876
- handling example for alpha channel
  - https://github.com/strukturag/libheif/blob/master/examples/heif_enc.cc#L587

Reply:
Got it. @firewood and @pixelcort, as yoya san mentioned and https://github.com/ImageMagick/ImageMagick/issues/1361. The alpha channel (transparent) is not supported by ImageMagick now. Let me know your thoughts.